tools/Tools.Common/Loaders/CmdletLoader.cs

```diff
@@ -208,9 +193,40 @@ public ModuleMetadata GetModuleMetadata(string assemblyPath)
             {
                 throw ex;
             }
+            finally
+            {
+                AssemblyLoadContext.Default.Resolving -= CustomResolving;
+            }
 
             ModuleMetadata.Cmdlets = results;
             return ModuleMetadata;
         }
+
+        private Assembly CustomResolving(AssemblyLoadContext assemblyLoadContext, AssemblyName name) => _alc.LoadFromAssemblyName(name);
+    }
+
+    internal class CmdletLoaderAlc : AssemblyLoadContext
+    {
+        private readonly List<string> _assemblyDirectories;
+
+        public CmdletLoaderAlc(List<string> assemblyDirectories = null)
+        {
+            _assemblyDirectories = assemblyDirectories ?? new List<string>();
+        }
+
+        protected override Assembly Load(AssemblyName assemblyName)
+        {
+            foreach (var assemblyPath in _assemblyDirectories)
+            {
+                var dll = Directory.GetFiles(assemblyPath, "*.dll").FirstOrDefault(f => Path.GetFileNameWithoutExtension(f) == assemblyName.Name);
+                if (dll == null)
+                {
+                    continue;
+                }
+
+                return LoadFromAssemblyPath(dll);
```

Comment:
The assembly resolving logic is the same as before. I just use `AssemblyLoadContext.LoadFromAssemblyPath()` to load the dll into individual ALCs